includes/wc-core-functions.php

```diff
@@ -722,19 +722,32 @@ function wc_get_image_size( $image_size ) {
 		'crop'   => 1,
 	);
 
+	// Size mapping.
+	if ( in_array( $image_size, array( 'shop_single', 'woocommerce_single' ), true ) ) {
+		$image_size = 'single';
+	} elseif ( in_array( $image_size, array( 'shop_thumbnail', 'shop_catalog', 'woocommerce_thumbnail' ), true ) ) {
+		$image_size = 'thumbnail';
+	}
+
 	if ( is_array( $image_size ) ) {
 		$size       = array(
 			'width'  => isset( $image_size[0] ) ? absint( $image_size[0] ) : 600,
 			'height' => isset( $image_size[1] ) ? absint( $image_size[1] ) : 600,
 			'crop'   => isset( $image_size[2] ) ? absint( $image_size[2] ) : 1,
 		);
 		$image_size = $size['width'] . '_' . $size['height'];
-	} elseif ( in_array( $image_size, array( 'single', 'shop_single', 'woocommerce_single' ), true ) ) {
+
+	} elseif ( 'single' === $image_size ) {
 		$size['width']  = absint( wc_get_theme_support( 'single_image_width', get_option( 'woocommerce_single_image_width', 600 ) ) );
 		$size['height'] = '';
 		$size['crop']   = 0;
-		$image_size     = 'single';
-	} elseif ( in_array( $image_size, array( 'thumbnail', 'shop_thumbnail', 'shop_catalog', 'woocommerce_thumbnail' ), true ) ) {
+
+	} elseif ( 'gallery_thumbnail' === $image_size ) {
+		$size['width']  = absint( wc_get_theme_support( 'gallery_thumbnail_image_width', 100 ) );
+		$size['height'] = $size['width'];
```

Comment:
The intention is to make the image square and this is not a typo, correct?

Reply:
Correct.

Reply:
Will this prevent themes from doing non-square thumbnails though if they want to?

Reply:
Themes can still modify the returned size but we don't allow themes to specify height for any of our sizes.

Reply:
How can I use non-square thumbnails? Is there a filter to overwrite the default square cropped ones?